src/server/actions/Volunteer.ts

```diff
@@ -41,9 +43,57 @@ export async function updateVolunteer(
       updatedVolunteer
     );
   } catch (error) {
-    //throw new CMError(CMErrorType.InternalError);
+    throw new CMError(CMErrorType.InternalError);
   }
   if (!res) {
-    // throw new CMError(CMErrorType.NoSuchKey, 'Volunteer');
+    throw new CMError(CMErrorType.NoSuchKey, 'Volunteer');
+  }
+}
+
+/**
+ * Get total hours that a volunteer has volunteered
+ * @param volunteerId // Id of the volunteer
+ * @returns // Total Hours of Volunteer
+ */
+export async function getVolunteerTotalHours(volunteerId: string) {
+  let volunteer;
+  let totalTime = 0;
+  try {
+    await dbConnect();
+    volunteer = await EventVolunteerSchema.find({
+      volunteer: volunteerId,
+    }).lean();
+
+    volunteer.forEach((ev) => {
+      if (ev.checkOutTime && ev.checkInTime) {
+        totalTime += ev.checkOutTime.getTime() - ev.checkInTime.getTime();
+      }
+    });
+  } catch (error) {
+    throw new CMError(CMErrorType.InternalError);
+  }
+  if (!volunteer) {
+    throw new CMError(CMErrorType.NoSuchKey, 'Volunteer');
+  }
+  // converting miliseconds to hours
+  return totalTime / 3600000;
+}
+
+/**
+ * Deletes a volunteer from all their events first then themselves.
+ * @param volunteerId // Id of the volunteer
+ * @returns // Deleted Volunteer
+ */
+export async function deleteVolunteer(volunteerId: string) {
+  try {
+    await dbConnect();
+
+    EventVolunteerSchema.deleteMany({
```

Comment:
This function looks good, but I believe you need an await on this line.

Reply:
Await added!